Refactor condor ID detection, reorganize gratia libs, bump version (SOFTWARE-5531)

brianhlin opened the pull request:
In SOFTWARE-5531, we decided to chown dirs properly in %post and I didn't want to reimplement the logic for how we determine the condor IDs. I factored out ID detection into a separate function and reworked the tests. To actually be able to call that function from outside of the script, I moved everything into a lib under `gratia-probe-common` and had the scripts just wrap around to call it.

You'll probably want to look at the commits outside of the rename for cleaner diffs.

brianhlin committed:
Separate functions for condor ID detection and priv dropping

(SOFTWARE-5531)

This will let us determine the appropriate UID/GID in %post.

Also, ensure that patches get torn down regardless of exceptions that
get raised in tests/setUp.

brianhlin committed:
Rename condor_meter for easier import (SOFTWARE-5531)

Also rename tests to better reflect their intention

brianhlin committed:
Move responsibility of providing probe name to the caller

e.g., condor_meter

brianhlin committed:
Minimal condor_meter that calls out to the library (SOFTWARE-5531)

brianhlin committed:
Update test imports to reflect rename (SOFTWARE-5531)

brianhlin committed:
Remove executable bit from the condor library (SOFTWARE-5531)

brianhlin committed:
Bump version to 2.8.4; chown condor dirs on %post (SOFTWARE-5531)

matyasselmeci commented on condor-ap/condor_meter:
```diff
@@ -0,0 +1,23 @@
+    except SystemExit:
+        raise
+    except KeyboardInterrupt:
+        raise
```
This can be deleted: as of Python 2.6 IIRC, SystemExit and KeyboardInterrupt derive from BaseException, not Exception.

matyasselmeci commented on rpm/gratia-probe.spec:
```diff
@@ -449,6 +456,12 @@ The HTCondor-CE probe for the Gratia OSG accounting system
+# Fix directory ownership if users have run condor_meter as root (SOFTWARE-5531)
+condor_ids=$(python3 -c 'import gratia.common.condor as condor; \
+             print(":".join([str(x) for x in core.get_condor_ids("htcondor-ce")]))')
+for dirname in $(awk -F '=' '/Folder=/ {print $2}' /etc/gratia/htcondor-ce/ProbeConfig | tr -d \"); do
```
This is terrifying. If someone messes up their ProbeConfig then the next time Gratia gets upgraded it will recursively chown whatever they put in there, which could be very destructive.

You should chown the default directories only (and declare ownership of them in the %files list if you haven't already); if the admin decides to move them elsewhere, that's their problem.

brianhlin replied:
What's the worst that could happen 😈 . Fixed.

matyasselmeci reviewed (changes requested):
See my comment about chowning dirs in %post.

brianhlin committed:
SystemExit and KeyboardInterrupt inherit from BaseException

So they won't be caught by `except Exception`

brianhlin committed:
Style fix-up

brianhlin committed:
Only change ownership on default directories (SOFTWARE-5531)

We don't want to muck around with dirs out of config in case
ProbeConfig is misconfigured
